README.md

```diff
@@ -5,40 +5,175 @@ Please edit this README.md file with your answers to the worksheet questions.
+
 
 ### c
+| A | B | C | A AND B | A AND C | (A AND B) OR (A AND C) |
+|---|---|---|---------|---------|------------------------|
+| 0 | 0 | 0 | 0       | 0       | 0                      |
+| 0 | 1 | 0 | 0       | 0       | 0                      |
+| 1 | 0 | 0 | 0       | 0       | 0                      |
+| 1 | 1 | 0 | 1       | 0       | 1                      |
+| 0 | 0 | 1 | 0       | 0       | 1                      |
+| 0 | 1 | 1 | 0       | 1       | 1                      |
+| 1 | 0 | 1 | 0       | 1       | 1                      |
+| 1 | 1 | 1 | 1       | 1       | 1                      |
+
+| A | B | C | B OR C | A AND (B OR C) |
+|---|---|---|--------|----------------|
+| 0 | 0 | 0 |   0    | 0              |
+| 0 | 1 | 0 |   1    | 0              |
+| 1 | 0 | 0 | 0      | 0              |
+| 1 | 1 | 0 | 1      | 1              |
+| 0 | 0 | 1 | 1      | 0              |
+| 0 | 1 | 1 | 1      | 0              |
+| 1 | 0 | 1 | 1      | 1              |
+| 1 | 1 | 1 | 1      | 1              |
+
 
 ### d
+| A | B | C | A OR B | A OR C | (A OR B) AND (A OR C) |
+|---|---|---|--------|--------|-----------------------|
+| 0 | 0 | 0 | 0      | 0      | 0                     |
+| 0 | 1 | 0 | 1      | 0      | 0                     |
+| 1 | 0 | 0 | 1      | 1      | 1                     |
+| 1 | 1 | 0 | 1      | 1      | 1                     |
+| 0 | 0 | 1 | 0      | 1      | 0                     |
+| 0 | 1 | 1 | 1      | 1      | 1                     |
+| 1 | 0 | 1 | 1      | 1      | 1                     |
+| 1 | 1 | 1 | 1      | 1      | 1                     |
+
+| A | B | C | B AND C | A OR (B AND C) |
+|---|---|---|---------|----------------|
+| 0 | 0 | 0 | 0       | 0              |
+| 0 | 1 | 0 | 0       | 0              |
+| 1 | 0 | 0 | 0       | 1              |
+| 1 | 1 | 0 | 0       | 1              |
+| 0 | 0 | 1 | 0       | 0              |
+| 0 | 1 | 1 | 1       | 1              |
+| 1 | 0 | 1 | 0       | 1              |
+| 1 | 1 | 1 | 1       | 1              |
 
 ## Question 4
 
 ### a
+This is the same identity as seen in Question 3B.
+"if not ( file_exists ("a . txt ") and file_exists ( "b. txt " )):" is equivalent to "NOT (A and B)".
+
+"if not file_exists ("a . txt ") or not file_exists (" b. txt " ):" is equivalent to "NOT A OR NOT B"
+
 
 ### b
+This is the same as can be seen in Question 3C. 
+The first statement is equivalent to sayings (B AND A) OR (C AND A) which can be reordered to be (A AND B) OR (A AND C).
+
 
 ### c
+This is the same as the question for 3A, for 3A if neither A or B are true then the code will execute the function "do_something()". This shows in the truth table for 3A as both A and B need to be false for the outputs to be true.
+
 
 ### d
+These programs are equivalent to that of Question 3D . The first section of code relates to the second part of the identity A or (B AND C) and the second section part of code relates to (A OR B) AND (A OR C).  
```

Comment:
Right idea, though the identity is a little off -- the second identity would be `B AND (A OR C)`